torchvision/csrc/io/image/cpu/decode_gif.cpp

```diff
@@ -0,0 +1,136 @@
+#include "decode_gif.h"
+#include "giflib/gif_lib.h"
+
+namespace vision {
+namespace image {
+
+typedef struct reader_helper_t {
+  torch::Tensor const* encoded_bytes;
```

Comment:
Can't we use a plain uint8 data pointer here instead of a pointer on the tensor?
Cast op in `read_from_tensor` looks a bit scary:
```
(reader_helper_t*)gifFile->UserData;
```

Reply:
I had that originally but then we need another field in `reader_helper_t` to keep track of the tensor's size (see how we access `(int)reader_helper->encoded_bytes->numel()`.

I don't really have a pref, LMK which one you prefer?

>  Cast op in read_from_tensor looks a bit scary

Just curious why you think that?